src/DeviceController.cpp

```diff
@@ -92,6 +93,19 @@ namespace earlyapp
                 GstVideoDevice::getInstance() : pVid;
             m_pCam = (pVid == nullptr) ?
                 GstCameraDevice::getInstance() : pCam;
+	     
+	    if(m_pConf->useCsicam())
```

Comment:
With new devices coming in, here will require more and more ugly if-else statements which makes this difficult to maintain. Why don't you consider a factory pattern that support multiple camera configurations?